drivers/SmartThings/matter-lock/src/init.lua

```diff
@@ -107,6 +116,23 @@ local function handle_battery_charge_level(driver, device, ib, response)
   end
 end
 
+local function handle_power_source_attribute_list(driver, device, ib, response)
```

Comment:
I think this is a good idea to use this method of calling `device:try_update_metadata({profile = profile_name})` in the attribute list handler itself, which will ideally avoid any of the race conditions I mentioned.

I think you'll want to use this same idea in the handlers for the other drivers as well, as it looks like it's missing in a few of them.

Reply:
Can you also add an explicit check for the battery level in this function too? I know that attribute is mandatory IF the battery feature is supported, but it'd still be good to be explicit about it here to prevent any bugs in the event that this attribute list handler would be used in the future. Something like this:
```
if attr.value == 0x0C then
  support_battery_percentage = true
end
if attr.value == 0x0E then
    support_battery_level = true
end
------then later add this :-----------
  if support_battery_percentage then
    profile_name = profile_name .. "-battery"
  elseif support_battery_level
    profile_name = profile_name .. "-batteryLevel"
  end